pyclesperanto_prototype/_tier0/_pycl.py

```diff
@@ -99,43 +99,50 @@ def assert_bufs_type(mytype, *bufs):
             % (mytype, str([b.dtype.type for b in bufs]))
         )
 
+def prepare(arr):
+    return np.require(arr, None, "C")
 
-def _wrap_OCLArray(cls):
-    """
-    WRAPPER
-    """
+class OCLArray(array.Array, np.lib.mixins.NDArrayOperatorsMixin):
 
-    def prepare(arr):
-        return np.require(arr, None, "C")
+    def __init__(self, queue, shape, dtype, order="C", allocator=None,
+                 data=None, offset=0, strides=None, events=None, _flags=None,
+                 _fast=False, _size=None, _context=None, _queue=None):
+        super().__init__(queue, shape, dtype, order, allocator,
+                     data, offset, strides, events, _flags,
+                     _fast, _size, _context, _queue)
```

Comment:
```suggestion
```
if you're simply calling the `super()` method, you don't need to reimplement it

Reply:
Ha! That's obviously different in Javd :-D